interoperability_report.py

```diff
@@ -0,0 +1,655 @@
+#!/usr/bin/python
+
+from multiprocessing import Process, Queue
+import time
+import re
+import pexpect
+import multiprocessing
+import argparse 
+import os
+from junitparser import TestCase, TestSuite, JUnitXml, Error, Attr
+from datetime import datetime
+
+from utilities import ReturnCode, path_executables
+from testSuite import dict_param_expected_code_timeout
+
+
+def subscriber(name_executable, parameters, key, time_out, code, data,
+                subscriber_finished, publisher_finished):
+    """ Run the executable with the parameters and save 
+        the error code obtained
+
+        name_executable     : name of the ShapeApplication to run
+                              as a Subscriber
+        parameters          : QOS to use with the Shape Application
+        key                 : test is being tested 
+                             (from dict_param_expected_code_timeout)
+        time_out            : time pexpect waits until it finds a pattern
+        code                : this variable will be overwritten with 
```

Comment:
no, it is not an output variable (it could not be because the child process cannot return anything to the main process). In order to do it i created a variable in the main process and i passed it to the child as a parameter in the function. Then the child overwrites the variable and the main process can access to it. This is done with 
`
manager = multiprocessing.Manager();
code = manager.list(range(2))
`
It creates a list of range 2, the first cell to save the Subscriber obtained ReturnCode, and the second one for the Publisher.

With the variable data it happens something similar, but the variable is a multiprocessing Queue. The Publisher will save the samples in the variable data and the Subscriber will take them from it.